core/trino-main/src/main/java/io/trino/operator/DeduplicatingDirectExchangeBuffer.java

```diff
@@ -141,7 +142,14 @@ public DeduplicatingDirectExchangeBuffer(
         }
 
         if (!outputReady.isDone()) {
-            return nonCancellationPropagating(outputReady);
+            return Futures.transformAsync(outputReady, ignored -> {
```

Comment:
Does this have to be `nonCancellationPropagating`?

Reply:
Yes, you are right. Just realized that outputReady may be cancelled too.